pkg/proxy/source_mock_test.go

```diff
@@ -29,8 +29,8 @@ type proxySourceMocker struct {
 	identity policy.NumericIdentity
 }
 
-func (m *proxySourceMocker) RLock()   { m.RLock() }
-func (m *proxySourceMocker) RUnlock() { m.RUnlock() }
+func (m *proxySourceMocker) RLock()   { m.RWMutex.RLock() }
+func (m *proxySourceMocker) RUnlock() { m.RWMutex.RUnlock() }
```

Comment:
This change seems unnecessary.

Reply:
As per @aanm this was needed for one of the tests failing. ?

Reply:
@tgraf it is necessary, otherwise the RLock() was a recursive function. :)